python/ray/air/util/data_batch_conversion.py

```diff
@@ -139,30 +138,22 @@ def _convert_batch_type_to_numpy(
                 )
         return data
     elif pyarrow is not None and isinstance(data, pyarrow.Table):
-        if data.column_names == [TENSOR_COLUMN_NAME] and (
-            isinstance(data.schema.types[0], ArrowTensorType)
-        ):
-            # If representing a tensor dataset, return as a single numpy array.
-            # Example: ray.data.from_numpy(np.arange(12).reshape((3, 2, 2)))
-            # Arrow’s incorrect concatenation of extension arrays:
-            # https://issues.apache.org/jira/browse/ARROW-16503
-            return _concatenate_extension_column(data[TENSOR_COLUMN_NAME]).to_numpy(
-                zero_copy_only=False
-            )
-        else:
-            output_dict = {}
-            for col_name in data.column_names:
-                col = data[col_name]
-                if col.num_chunks == 0:
-                    col = pyarrow.array([], type=col.type)
-                elif _is_column_extension_type(col):
-                    # Arrow’s incorrect concatenation of extension arrays:
-                    # https://issues.apache.org/jira/browse/ARROW-16503
-                    col = _concatenate_extension_column(col)
-                else:
-                    col = col.combine_chunks()
-                output_dict[col_name] = col.to_numpy(zero_copy_only=False)
-            return output_dict
+        output_dict = {}
+        for col_name in data.column_names:
+            col = data[col_name]
+            if col.num_chunks == 0:
+                col = pyarrow.array([], type=col.type)
+            elif _is_column_extension_type(col):
+                # Arrow’s incorrect concatenation of extension arrays:
+                # https://issues.apache.org/jira/browse/ARROW-16503
+                col = _concatenate_extension_column(col)
```

Comment:
I'm about to merge a fix to this part of `ArrowBlock.to_numpy()` btw, I found that `pyarrow.array([], type=col.type)` does not work for the `col.num_chunks == 0` case when the column is an extension array, so we need to reorder these and have `_concatenate_extension_column` properly handle the `col.num_chunks == 0` for the extension array case. Relevant bits of the PR about to merged are here:
- https://github.com/ray-project/ray/pull/29055/files#diff-a637aa9db5408fe440f8b335a480e66e6636d97e71dc4bad164c356ac43deafdR211-R214
- https://github.com/ray-project/ray/pull/29055/files#diff-bf26e743f5dc78351cf2755d92da8f54a0e1d37baa8b809dc679e64ddce83840

You could port these to your PR or wait until I merged (in the next hour or two) and then rebase.